### DISCUSSION TITLE
Refresh Relationship Picker as an Action

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#5524] [Help]
Is there a way, to refresh a Relationship Picker with an Action ?

Scenario : 
I have Doctor Specialties 1 - > Many with Doctors
I go to Add new Doctor Screen
The Specialty I want to assign doesn't exist 
Click Add Specialty button
Button opens a modal , user adds new Specialty and modal closes.

How do i refresh the Relationship Picker ?

<img width="734" alt="Screenshot 2022-04-21 at 8 03 22 PM" src="https://user-images.githubusercontent.com/80512170/164513992-61b23f71-308b-4fbd-a432-25b57a363a48.png">

**Comment by melohagan** [comment]
Hi @poirazis 

If you link the new specialty to an existing doctor, then it will refresh that specialty list. 

You can use URL variables: https://docs.budibase.com/docs/url-parameters to pass on the Doctor ID to the new specialty screen, and then use a default value:

### Doctor Detail Screen
![Screenshot 2022-04-22 at 10 36 53](https://user-images.githubusercontent.com/101575380/164679825-81d82d6d-1d3b-43d9-83cf-1da46a0a7d17.png)

### New Specialty Screen
![Screenshot 2022-04-22 at 10 33 39](https://user-images.githubusercontent.com/101575380/164679229-9f0719d3-a567-49a6-866e-6d2430890110.png)
![Screenshot 2022-04-22 at 10 34 46](https://user-images.githubusercontent.com/101575380/164679424-147c89d4-7004-42ba-92ea-24b1ebfd5e0d.png)
 
----

For a new doctor not yet saved, I played around with a few ideas, but it is a lot trickier. In particular because the 'add specialty' screen opens in a modal, I couldn't find a way to trigger a navigation after closing the modal. 
This would have effectively allowed a refresh of the page, with the default values of the form saved in App State: https://docs.budibase.com/docs/app-state

It might be nice to have an optional redirect URL for the close modal action. I've suggested that here: https://github.com/Budibase/budibase/discussions/1176#discussioncomment-2614757

**Reply by poirazis** [reply]
Thank you @melohagan !
